modules/openapi-generator/src/main/resources/C-libcurl/apiClient.c.mustache

```diff
@@ -632,7 +632,12 @@ char *base64encode (const void *b64_encode_this, int encode_this_many_bytes){
     BUF_MEM_grow(mem_bio_mem_ptr, (*mem_bio_mem_ptr).length + 1);   //Makes space for end null.
     (*mem_bio_mem_ptr).data[(*mem_bio_mem_ptr).length] = '\0';  //Adds null-terminator to tail.
     return (*mem_bio_mem_ptr).data; //Returns base-64 encoded data. (See: "buf_mem_st" struct).
-#endif
+#else // OPENSSL
+    char* ret = malloc(encode_this_many_bytes);
+    memcpy(ret, b64_encode_this, encode_this_many_bytes-1);
+    ret[encode_this_many_bytes] = 0;
+    return ret;
+#endif // OPENSSL
 }
```

Comment:
Why this fake encode/decode is needed ?  If uses does not define ```-DOPENSSL```, but use the function ```base64decode``` or ```base64encode``` ,  they will not get the result that they want.

I think if the users want to use the function ```base64decode``` or ```base64encode``` defined by openapi-generator/c-libcurl , they must define -DOPENSSL in CMakefile.

Reply:
I didn't get it. What if `-DOPENSSL` is not defined? What do you intend to happen in that case? I didn't have it defined, and the function "returned" without issuing a `return` statement.

Reply:
e.g. 
If uses do not define ```-DOPENSSL```, but use the function ```base64encode```, the returned ```char *``` is still equal to the original string, not a base64 encoded string.

Reply:
I thought that was idea, and that's the code I implemented. The original code did nothing (not even `return` a buffer).

Reply:
Yes. I know.
So I suggest printing a warning message here if this fake code is needed:
e.g.
```c
#else // OPENSSL
    fprintf(stderr, "The data will not be decoded. If you want to use this function, please define \"-DOPENSSL\" when building.\n");
...
    return ret;
#endif // OPENSSL
```

Reply:
ok, will do that

Reply:
Message proposal: 

The data will not be encoded. If you want to use this function \"base64encode\", please define \"-DOPENSSL\" when building the library.
The base64 data will not be decoded. If you want to use this function, \"base64decode\", please define \"-DOPENSSL\" when building the library.

Reply:
I would suggest returning NULL so the whole API call exits with an error.  this is because we use strlen to get the length of the data when we add data to curl handle. This will break if the data has binary(may include null termination). So for a safer side, I would suggest to return NULL or mandate OpenSSL installation as a prerequisite.

Reply:
What if I don't have OpenSSL, and I just want to generate unencrypted APIs?

Reply:
If you don't have OpenSSL then you cannot send or receive binary data. 

Also, I didn't get what you mean by unencrypted API? Because OpenSSL is used only for encoding binary data. It is not used for encryption.

Reply:
Ok.

Reply:
I updated also `CMakeLists.txt`, but I am not an expert on `cmake`.